### PR TITLE
pppoe-server: T5710: Add option permit any-login

### PR DESCRIPTION
**Pull request by natali-rs1985**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5710
## Component(s) name

pppoe-server
## How to test

```
set service pppoe-server authentication local-users username user1 password 'user1'
set service pppoe-server authentication mode 'local'
set service pppoe-server client-ip-pool first range '100.64.0.1-100.64.0.100'
set service pppoe-server default-pool 'first'
set service pppoe-server gateway-address '100.64.0.1'
set service pppoe-server interface eth1

set service pppoe-server authentication any-login
commit

vyos@vyos# cat /run/accel-pppd/pppoe.conf | grep any-login
any-login=1
```
## Smoketest result

```
vyos@vyos# python3 /usr/libexec/vyos/tests/smoke/cli/test_service_pppoe-server.py
test_accel_ipv4_pool (__main__.TestServicePPPoEServer.test_accel_ipv4_pool) ... ok
test_accel_ipv6_pool (__main__.TestServicePPPoEServer.test_accel_ipv6_pool) ... ok
test_accel_limits (__main__.TestServicePPPoEServer.test_accel_limits) ... ok
test_accel_local_authentication (__main__.TestServicePPPoEServer.test_accel_local_authentication) ... ok
test_accel_log_level (__main__.TestServicePPPoEServer.test_accel_log_level) ... ok
test_accel_name_servers (__main__.TestServicePPPoEServer.test_accel_name_servers) ... ok
test_accel_next_pool (__main__.TestServicePPPoEServer.test_accel_next_pool) ... ok
test_accel_ppp_options (__main__.TestServicePPPoEServer.test_accel_ppp_options) ... ok
test_accel_radius_authentication (__main__.TestServicePPPoEServer.test_accel_radius_authentication) ... ok
test_accel_shaper (__main__.TestServicePPPoEServer.test_accel_shaper) ... ok
test_accel_snmp (__main__.TestServicePPPoEServer.test_accel_snmp) ... ok
test_accel_wins_server (__main__.TestServicePPPoEServer.test_accel_wins_server) ... ok
test_pppoe_limits (__main__.TestServicePPPoEServer.test_pppoe_limits) ... ok
test_pppoe_server_any_login (__main__.TestServicePPPoEServer.test_pppoe_server_any_login) ... ok
test_pppoe_server_authentication_protocols (__main__.TestServicePPPoEServer.test_pppoe_server_authentication_protocols) ... ok
test_pppoe_server_pado_delay (__main__.TestServicePPPoEServer.test_pppoe_server_pado_delay) ... ok
test_pppoe_server_shaper (__main__.TestServicePPPoEServer.test_pppoe_server_shaper) ... ok
test_pppoe_server_vlan (__main__.TestServicePPPoEServer.test_pppoe_server_vlan) ... ok

----------------------------------------------------------------------
Ran 18 tests in 113.655s

OK
```
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by sever-sever — APPROVED** [review]

File: interface-definitions/service_pppoe-server.xml.in [hunk 1]
```diff
@@ -47,6 +47,12 @@
                   </leafNode>
                 </children>
               </node>
+              <leafNode name="any-login">
```
Comment (c-po) [inline interface-definitions/service_pppoe-server.xml.in:50]: Why not call this `no-auth` like accell is doing?

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by c-po** [comment]
Please add a smoketest validating the new CLI option

File: interface-definitions/service_pppoe-server.xml.in (hunk 1 above)
Comment (natali-rs1985) [reply interface-definitions/service_pppoe-server.xml.in:50]: this is a different option.
from the documentation:
```
"any-login=" 0\|1
If set to 1 allows pass PAP, CHAP and Microsoft CHAP v1 authentication with any login.

"noauth=" 0\|1
If set to 1 skips any PPP and IPOE authentication.
```

**Comment by natali-rs1985** [comment]
> Please add a smoketest validating the new CLI option

Smoketest was added

**Review by c-po — APPROVED** [review]

**Pull request merged**